### PR TITLE
Added Terraform Rules for https://github.com/prancer-io/prancer-compliance-test/issues/174 & More (Details in the description)

### PR DESCRIPTION
**Pull request by rezoan**
- Added Terraform Rules for https://github.com/prancer-io/prancer-compliance-test/issues/174
- fixed aks aad RBAC rule to cope with the block array structure after conversion to json.
- Verified and fixed rego rules and test case for Terraform aks.rego
- verified and fixed existing rego rules and test case for Terraform applicationgateways.rego
- verfied and fixed existing rego rules and test case for Terraform dbadministrator.rego
- verified and fixed existing rego rules and test case for Terraform dbauditingsettings.rego
- verified and fixed existing rego rules and test case for Terraform dbserverauditingsettings.rego
- verified and fixed existing rego rules and test case for Terraform dbfirewallrules.rego
- verified and fixed existing rego rules and test case for Terraform dbsecurityalertpolicies.rego
- verified and fixed existing rego rules and test case for Terraform dbvulnerabilityassessments.rego
- verified and fixed existing rego rules and test case for Terraform keyvaultsecrets.rego
- verified and fixed existing rego rules and test case for Terraform verified pricing.rego

### CODE REVIEW

File: azure/terraform/databricks.rego [hunk 1]
```diff
@@ -0,0 +1,119 @@
+package rule
+
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/databricks_workspace
+
+#
+# PR-AZR-0070-TRF
```
Comment (farchide) [inline azure/terraform/databricks.rego:6]: @rezoan  please use the same number for the Azure Databricks in ARM template and in Terraform

Comment (rezoan) [reply azure/terraform/databricks.rego]: done.

File: azure/terraform/dbadministrators.rego [hunk 2]
```diff
@@ -20,39 +20,37 @@ azure_issue["db_ad_admin"] {
     lower(resource.type) == "azurerm_sql_server"
     count([c | r := input.resources[_];
                r.type == "azurerm_sql_active_directory_administrator";
-               re_match(concat("", ["^.*\\.", resource.name, "\\..*$"]), r.properties.server_name);
+               re_match(concat("", ["^.*\\.", resource.name, "\\..*$"]), r.properties.server_name); # Rezoan: this regex is not correct and need a fix. Snapshot file does not generate server_name out of the tf variable as well at r.properties.server_name
```
Comment (farchide) [inline azure/terraform/dbadministrators.rego:23]: @rezoan assign a task to @vatsalgit5118 to fix the generated snapshot

**Review by farchide — APPROVED** [review]

File: azure/terraform/dbadministrators.rego (hunk 2 above)
Comment (rezoan) [reply azure/terraform/dbadministrators.rego:23]: @farchide bugs has already been reported 
- https://github.com/prancer-io/prancer-compliance-test/issues/138
- https://github.com/prancer-io/prancer-compliance-test/issues/139

**Pull request merged**